ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -858,84 +981,66 @@ namespace OrdinalAxis {
-                    linearEquivalentRight,
-                    distance;
-
-
-                // Handle the case where we translate from the index directly,
-                // used only when panning an ordinal axis
-                if (fromIndex) {
-
-                    if (val < 0) { // out of range, in effect panning to the left
-                        val = ordinalPositions[0];
-                    } else if (val > i) { // out of range, panning to the right
-                        val = ordinalPositions[i];
-                    } else { // split it up
-                        i = Math.floor(val);
-                        distance = val - i; // the decimal
-                    }
+                // Check if the index is inside position array.
+                // If true, read/approximate value for that exact index.
+                if (indexInPostions >= 0 && indexInPostions < positions.length) {
+                    const leftNeighbour = positions[Math.floor(indexInPostions)],
+                        rightNeighbour = positions[Math.ceil(indexInPostions)],
+                        distance = rightNeighbour - leftNeighbour;
 
-                // Loop down along the ordinal positions. When the linear
-                // equivalent of i matches an ordinal position, interpolate
-                // between the left and right values.
-                } else {
-                    while (i--) {
-                        linearEquivalentLeft =
-                            ((ordinalSlope as any) * i) + (ordinalOffset as any);
-                        if (val >= linearEquivalentLeft) {
-                            linearEquivalentRight =
-                                ((ordinalSlope as any) *
-                                (i + 1)) +
-                                (ordinalOffset as any);
-                            // something between 0 and 1
-                            distance = (val - linearEquivalentLeft) /
-                                (linearEquivalentRight - linearEquivalentLeft);
-                            break;
-                        }
-                    }
+                    return positions[Math.floor(indexInPostions)] + mantissa * distance;
                 }
 
-                // If the index is within the range of the ordinal positions,
-                // return the associated or interpolated value. If not, just
-                // return the value.
-                return (
-                    typeof distance !== 'undefined' &&
-                    typeof ordinalPositions[i] !== 'undefined' ?
-                        ordinalPositions[i] + (
-                            distance ?
-                                distance *
-                                (ordinalPositions[i + 1] - ordinalPositions[i]) :
-                                0
-                        ) :
-                        val
-                );
+                // For cases when the index is not in the EOP array,
```

Comment:
A little nitpicking here, sorry!

What is `EOP` array? It would be nice in general to avoid abbreviations. Does it make sense to name variables simply `index` and `length`? 

Similar question to `firstEOPPoint`. In `positions` we have x-values, right? Not points (as Point instance classes)